internal/cli/cmd/connectivity.go

```diff
@@ -47,74 +49,89 @@ var params = check.Parameters{
 
 var tests []string
 
+func RunE(hooks Hooks) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		params.CiliumNamespace = namespace
+
+		for _, test := range tests {
+			if strings.HasPrefix(test, "!") {
+				rgx, err := regexp.Compile(strings.TrimPrefix(test, "!"))
+				if err != nil {
+					return fmt.Errorf("test filter: %w", err)
+				}
+				params.SkipTests = append(params.SkipTests, rgx)
+			} else {
+				rgx, err := regexp.Compile(test)
+				if err != nil {
+					return fmt.Errorf("test filter: %w", err)
+				}
+				params.RunTests = append(params.RunTests, rgx)
+			}
+		}
+
+		// Instantiate the test harness.
+		cc, err := check.NewConnectivityTest(k8sClient, params, version)
+		if err != nil {
+			return err
+		}
+
+		ctx, _ := signal.NotifyContext(cmd.Context(), os.Interrupt, syscall.SIGTERM)
+
+		if params.Timeout > 0 {
+			timeoutCtx, cancelFunc := context.WithTimeoutCause(ctx, params.Timeout, fmt.Errorf("connectivity test suite timeout (%s) reached", params.Timeout))
+			defer cancelFunc()
+			ctx = timeoutCtx
+		}
+
+		go func() {
+			<-ctx.Done()
+			cc.Logf("Cancellation request (%s) received, cancelling tests...", context.Cause(ctx))
+		}()
+
+		done := make(chan struct{})
+		var finished bool
+
+		// Execute connectivity.Run() in its own goroutine, it might call Fatal()
+		// and end the goroutine without returning.
+		go func() {
+			defer func() { done <- struct{}{} }()
+			err = connectivity.Run(ctx, cc, hooks.AddConnectivityTests, hooks.SetupAndValidate)
+
+			// If Fatal() was called in the test suite, the statement below won't fire.
+			finished = true
+		}()
+		<-done
+
+		if !finished {
+			// Exit with a non-zero return code.
+			return errInternal
+		}
+
+		if err != nil {
+			return fmt.Errorf("connectivity test failed: %w", err)
+		}
+
+		return nil
+	}
+}
+
+func RunEPerf(hooks Hooks) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		// This is ugly hack that allows us to override default values
+		// of these parameters that are not visible in perf subcommand options
+		// as we can't have different defaults specified in test and perf subcommand
+		params.Perf = true
+		params.ForceDeploy = true
+		return RunE(hooks)(cmd, args)
+	}
+}
+
```

Comment:
There's a way we can get rid of this hack. I'm pretty sure you could just use a [PreRun](https://github.com/spf13/cobra/blob/v1.8.0/command.go#L129) function to modify the params instead.

Reply:
Great idea, done!